src/xunit.performance.api/SafeTerminateHandler.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.Xunit.Performance.Api.Native.Windows;
+using System;
+
+namespace Microsoft.Xunit.Performance.Api
+{
+    internal sealed class SafeTerminateHandler<T> : IDisposable
+        where T : class, IDisposable
+    {
+        public SafeTerminateHandler(Func<T> createCallback)
+        {
+            BaseDisposableObject = null;
+
+            _HandlerRoutine = (Kernel32.CtrlTypes dwCtrlType) =>
+            {
+                switch (dwCtrlType)
+                {
+                    case Kernel32.CtrlTypes.CTRL_C_EVENT:
+                    case Kernel32.CtrlTypes.CTRL_BREAK_EVENT:
+                    case Kernel32.CtrlTypes.CTRL_CLOSE_EVENT:
+                        BaseDisposableObject?.Dispose();
+                        break;
+                    case Kernel32.CtrlTypes.CTRL_LOGOFF_EVENT:
+                    case Kernel32.CtrlTypes.CTRL_SHUTDOWN_EVENT:
+                        break;
```

Comment:
Is it required to add these if they're not treated specifically? Can we use `default` instead?